example/plugin/toerror/toerror.go

```diff
@@ -0,0 +1,10 @@
+package toerror
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func getJsonTag(f reflect.StructTag) (string, error) {
```

Comment:
I love using reflect here, what a great example.
Maybe we can compose it with another function that takes a string and returns another value and error, to reflect your use case stated in the example, that this function is now easier to compose.
What do you think?